extensions/wikia/Answers/SpecialCreateDefaultQuestionPage.php

```diff
@@ -1,58 +1,108 @@
 <?php
 class CreateQuestionPage extends UnlistedSpecialPage {
 
+	const QS_KEY_QUESTION = 'questiontitle';
+
 	function __construct(){
 		parent::__construct("CreateQuestionPage");
 	}
 
-	function execute( $question ){
-		global $wgRequest, $wgOut, $wgUser;
+	function execute( $question ) {
+		if ( $this->getRequest()->wasPosted() ) {
+			return $this->createQuestion();
+		}
+
+		return $this->questionPage( $question );
+	}
+
+	private function questionPage( $question ) {
```

Comment:
Actually, I don't think we need this method at all actually.  Previously this code essentially did data validation and took the user somewhere appropriate if the data was missing or incomplete.  Here its been duplicated (I see the same lines still exist below) and used as if it were a valid alternate branch of `execute`, e.g., "If you `GET` this endpoint you should get a help page, if you `POST` we should create a new question"

What `wasPosted()` is actually testing for is a security concern.  If it fails that, the request should fail.